.github/workflows/release-dataset.yml

```diff
@@ -0,0 +1,72 @@
+  release_dataset: 
+    name: Release Dataset
+    runs-on: ubuntu-latest
+    env:
+      NODE_ENV: "production"
+      FIREBASE_TOKEN: ${{ secrets.FIREBASE_TOKEN }}
+      FIREBASE_EMAIL: ${{ secrets.FIREBASE_EMAIL }}
+      AWS_SECRET: ${{ secrets.CFE_AWS_SECRET }}
+      AWS_ID: ${{ secrets.CFE_AWS_ID }}
+    steps:
+      - uses: actions/checkout@v3
+      - name: Setup Node
+        uses: actions/setup-node@v3
+        with:
+          node-version: 16.x
+      - run: npm ci
+      - name: Run release-dataset script
+        run: |
+          if: ${{ github.event.inputs.megaset-name }} && ${{ github.event.inputs.contained-dataset-id }}
+            npm run release-dataset ${{ github.event.inputs.megaset-name }} ${{ github.event.inputs.contained-dataset-id }}
+          elif [ -n "${{ github.event.inputs.megaset-name }}" ]; then
+            npm run release-dataset ${{ github.event.inputs.megaset-name }}
+          elif [ -n "${{ github.event.inputs.standalone-dataset-id }}" ]; then
+            npm run release-dataset ${{ github.event.inputs.standalone-dataset-id }}
+          else
+            echo "Error: either megaset name or dataset ID must be specified"
+            exit 1
+          fi
+      - name: Release every dataset in a megaset
+        if: ${{ github.event.inputs.megaset-name == 'true' }} && ${{ github.event.inputs.contained-dataset-id == 'false' }}
+        run: npm run release-dataset ${{ github.event.inputs.megaset-name }}
+      - name: Release a dataset contained within a megaset
+        if: ${{ github.event.inputs.megaset-name == 'true' }} && ${{ github.event.inputs.contained-dataset-id == 'true' }}
+        run: npm run release-dataset ${{ github.event.inputs.megaset-name }} ${{ github.event.inputs.contained-dataset-id }}
+      - name: Release a dataset that is not part of a megaset
+        if: ${{ github.event.inputs.megaset-name == 'false' }} && ${{ github.event.inputs.standalone-dataset-id == 'true' }}
+        run: npm run release-dataset ${{ github.event.inputs.megaset-name }} ${{ github.event.inputs.standalone-dataset-id }}
+      - name: Error out if input variables are conflict   
+        if: ${{ github.event.inputs.megaset-name == 'true' }} && ${{ github.event.inputs.standalone-dataset-id == 'true' }}
+        run: | 
+          echo "Error: cannot specify both a megaset and a standalone dataset ID. Please provide only one of these inputs."
+          exit 1
+      - name: Error out if no input variables are specified 
+        if: ${{ !github.event.inputs.megaset-name && !github.event.inputs.contained-dataset-id && !github.event.inputs.standalone-dataset-id }}
+        run: | 
+          echo "Error: either megaset name, contained dataset ID, or standalone dataset ID must be specified"
+          exit 1
+      - name: Error out if no specified megaset for a contained dataset ID
+        if: ${{ github.event.inputs.megaset-name == 'false' }} && ${{ github.event.inputs.contained-dataset-id == 'true' }}
```

Comment:
this is too complex for a GitHub action, especially for this stage when you're testing

Reply:
I actually think so too... I'll try to figure out a cleaner way. Thank you for commenting!